### ISSUE TITLE
Array conversion doesn't work in different context

### ISSUE THREAD

**Issue by SilentGert** [nashwaan/xml-js#42]
Hi, 
In my application I run user scripts in a [Node VM](https://nodejs.org/api/vm.html). I added your library to provide easy XML parsing and generation capabilities. Now I run into a problem when I try to convert an `Array` to XML. The array is created within the user script which runs in a sandbox context. The library is added to this context, but uses the wrapping context itself. 

As far as I can tell, the problem are the `instanceof Array` checks in your conversion code. Because I have now two different versions of the `Array` object (one in each context) the check doesn't work anymore. See also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof

Reusing the same context is currently not possible for me as the scripts have to run in a sandbox with limited access.

I created a test case that illustrates the problem:
```javascript
var convert = require("../lib");
var Script = require("vm").Script;

describe("case by silentgert", function() {
      var context = {
        convert: convert,
        output: undefined,
      };
      var scriptCode = 
      "(function() {\n" +
      "  const obj = {\n" +
      "    customers : {\n" +
      "      customer: [\n" +
      "        {\n" +
      "          _text: 'John Doe',\n" +
      "          _attributes: {\n" +
      "            status: 'silver'\n" +
      "          }\n" +
      "        },\n" +
      "        {\n" +
      "          _text: 'Alice Allgood',\n" +
      "          _attributes: {\n" +
      "            status: 'gold'\n" +
      "          }\n" +
      "        }\n" +
      "      ]\n" +
      "    }\n" +
      "  };\n" +
      "  output = convert.js2xml(obj, { compact: true });\n" +
      "})()\n";

      var executableScript = new Script(scriptCode, {
        displayErrors: true,
      });
  
      it ('should convert Arrays in a different context', function() {
        executableScript.runInNewContext(context);
        expect(context.output).toEqual('<customers><customer status="silver">John Doe</customer><customer status="gold">Alice Allgood</customer></customers>');
      });
    });
```

Expect outcome:
```xml
<customers>
  <customer status="silver">John Doe</customer>
  <customer status="gold">Alice Allgood</customer>
</customers>
```

Observed outcome:
```xml
<customers>
  <customer>
    <0 status="silver">John Doe</0>
    <1 status="gold">Alice Allgood</1>
  </customer>
</customers>
```

Possible solution:
In the Mozilla docs they mention to use `Array.isArray` for secure checks. Another option might be to look at how Lodash is doing the `isArray` check (without adding the dependency). 

If you are okay with it, I can look into the issue and provide a pull request.

**Comment by nashwaan** [comment]
@SilentGert Sorry for very late reply.

> If you are okay with it, I can look into the issue and provide a pull request.

I would be glad to accept your pull request.

**Comment by nashwaan** [comment]
Pull #43 is merged and this is now published in version `1.6.2` of this library.

Thanks @SilentGert  for the investigations and putting the efforts into fixing the code.

👍👍👍👍👍

**Comment by SilentGert** [comment]
I'm happy to help. Thank you for providing the library. :)
